pandaserver/dataservice/event_picker.py

```diff
@@ -0,0 +1,276 @@
+        str_msg = self.logger.dumpToString()
+        status, output = Client.uploadLog(str_msg, self.jedi_task_id)
+        if status != 0:
+            return f"failed to upload log with {status}."
+        if output.startswith("http"):
+            return f'<a href="{output}">log</a>'
+        return output
+
+    def get_options_from_file(self) -> dict:
+        """
+        Gets options from the event picking file.
+
+        This method reads the event picking file and extracts options from it. The options are stored in a dictionary and returned.
+
+        Returns:
+            dict: A dictionary containing the options extracted from the event picking file.
+        """
+        options = {
+            "runEvent": [],
+            "eventPickDataType": "",
+            "eventPickStreamName": "",
+            "eventPickDS": [],
+            "eventPickAmiTag": "",
+            "eventPickNumSites": 1,
+            "inputFileList": [],
+            "tagDS": [],
+            "tagQuery": "",
+            "tagStreamRef": "",
+            "runEvtGuidMap": {},
+            "ei_api": "",
+        }
+
+        for tmp_line in self.event_picking_file:
+            tmp_match = re.search("^([^=]+)=(.+)$", tmp_line)
+            if tmp_match is not None:
+                key, value = tmp_match.groups()
+                if key in options:
+                    if key == "runEvent":
+                        options[key].append(value.split(","))
+                    elif key in ["eventPickDS", "inputFileList", "tagDS"]:
+                        options[key] = value.split(",")
+                    elif key == "eventPickNumSites":
+                        options[key] = int(value)
+                    else:
+                        options[key] = value
+        return options
+
+    def get_jedi_task_id(self, options: dict) -> int:
+        """
+        Gets the jediTaskID.
+
+        This method gets the jediTaskID from the task buffer using the user's DN and task name. It also updates the task modification time in the task buffer if the parameters contain "--eventPickSkipDaTRI".
+
+        Parameters:
+            options (dict): A dictionary containing the options extracted from the event picking file.
+
+        Returns:
+            int: The jediTaskID.
+        """
+        self.user_dn = options["userName"]
+        self.user_task_name = options["userTaskName"]
+        self.user_dataset_name = options["userDatasetName"]
+        self.locked_by = options["lockedBy"]
+        self.creation_time = options["creationTime"]
+        self.params = options["params"]
+
+        # extract task name
+        if self.user_task_name == "" and self.params != "":
+            tmp_match = re.search("--outDS(=| ) *([^ ]+)", self.params)
+            if tmp_match is not None:
+                self.user_task_name = tmp_match.group(2)
+                if not self.user_task_name.endswith("/"):
+                    self.user_task_name += "/"
+
+        # suppress DaTRI
```

Comment:
CC @tmaeno 
Is this needed? DaTRI doesn't exist anymore. I ran `zgrep "skip DaTRI" panda-EventPicker.log*` on panda server and there were no results.